### PR TITLE
Replace many loops with range-for

### PR DESCRIPTION
**Pull request by lgritz**
As suggested by clang-tidy

### CODE REVIEW

File: src/liboslcomp/codegen.cpp [hunk 1]
```diff
@@ -98,8 +98,7 @@ OSLCompilerImpl::insert_code (int opnum, const char *opname,
     // the jump addresses of other ops and the param init ranges.
     if (opnum < (int)m_ircode.size()-1) {
         // Adjust jump offsets
-        for (size_t n = 0;  n < m_ircode.size();  ++n) {
-            Opcode &c (m_ircode[n]);
+        for (auto & c : m_ircode) {
```
Comment (fpsunflower) [inline src/liboslcomp/codegen.cpp]: Whitespace ...

File: src/liboslcomp/oslcomp.cpp [hunk 2]
```diff
@@ -945,37 +945,37 @@ OSLCompilerImpl::write_oso_file (const std::string &outfilename,
     int lastline = -1;
     ustring lastfile;
     ustring lastmethod ("___uninitialized___");
-    for (OpcodeVec::iterator op = m_ircode.begin(); op != m_ircode.end();  ++op) {
-        if (lastmethod != op->method()) {
-            oso ("code %s\n", op->method().c_str());
-            lastmethod = op->method();
+    for (auto & op : m_ircode) {
```
Comment (fpsunflower) [inline src/liboslcomp/oslcomp.cpp]: And here too.

**Comment by fpsunflower** [comment]
LGTM!

File: src/liboslcomp/codegen.cpp (hunk 1 above)
Comment (lgritz) [inline src/liboslcomp/codegen.cpp]: D'oh!  That's what `clang-tidy --fix` did. I tried to fix them all by hand, looks like I missed one or two. I'll fix.

**Comment by lgritz** [comment]
Updated with two fewer spaces. :-)

**Comment by fpsunflower** [comment]
LGTM!

**Pull request merged**